packages/flutter/test/widgets/semantics_test.dart

```diff
@@ -828,7 +831,133 @@ void main() {
             ],
           ),
         ],
-      ), ignoreTransform: true, ignoreRect: true, ignoreId: true));
+      ), ignoreTransform: true, ignoreRect: true, ignoreId: true),
+    );
+    semantics.dispose();
+  });
+
+  testWidgets('Semantics widgets without sort orders are sorted properly', (WidgetTester tester) async {
+    final SemanticsTester semantics = new SemanticsTester(tester);
+    int semanticsUpdateCount = 0;
+    tester.binding.pipelineOwner.ensureSemantics(
+      listener: () {
+        semanticsUpdateCount += 1;
+      }
+    );
+    await tester.pumpWidget(
+      new Directionality(
+        textDirection: TextDirection.ltr,
+        child: new Column(
+          children: <Widget>[
+            const Text('Label 1'),
+            const Text('Label 2'),
+            new Row(
+              children: <Widget>[
+                const Text('Label 3'),
+                const Text('Label 4'),
+                const Text('Label 5'),
+              ],
+            ),
+          ],
+        ),
+      ),
+    );
+    expect(semanticsUpdateCount, 1);
+    expect(semantics, hasSemantics(
+      new TestSemantics(
+        children: <TestSemantics>[
+          new TestSemantics(
+            label: r'Label 1',
+            textDirection: TextDirection.ltr,
+            nextNodeId: 3,
+          ),
+          new TestSemantics(
+            label: r'Label 2',
+            textDirection: TextDirection.ltr,
+            nextNodeId: 4,
+          ),
+          new TestSemantics(
+            label: r'Label 3',
+            textDirection: TextDirection.ltr,
+            nextNodeId: 5,
+          ),
+          new TestSemantics(
+            label: r'Label 4',
+            textDirection: TextDirection.ltr,
+            nextNodeId: 6,
+          ),
+          new TestSemantics(
+            label: r'Label 5',
+            textDirection: TextDirection.ltr,
+            nextNodeId: -1,
+          ),
+        ],
+      ), ignoreTransform: true, ignoreRect: true, ignoreId: true),
+    );
+    semantics.dispose();
+  });
+
+  testWidgets('Semantics widgets that are transformed are sorted properly', (WidgetTester tester) async {
+    final SemanticsTester semantics = new SemanticsTester(tester);
+    int semanticsUpdateCount = 0;
+    tester.binding.pipelineOwner.ensureSemantics(
+      listener: () {
+        semanticsUpdateCount += 1;
+      }
+    );
+    await tester.pumpWidget(
+      new Directionality(
```

Comment:
what if there's no Directionality widget? e.g. with a Stack rather than a Column.

Reply:
You can use a stack without directionality, just set its alignment to something non-directional.

Reply:
OK, see below.